### PR TITLE
Debugging support using tmate

### PR DESCRIPTION
**Pull request by shubham149**

### CODE REVIEW

File: engine/compiler/compiler.go [hunk 1]
```diff
@@ -427,6 +437,35 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
...
+	// create internal steps if build running in debug mode
+	if c.Tmate.Enabled && args.Build.Debug && pipeline.Platform.OS != "windows" {
+		// first we need to add an internal setup step to the pipeline
+		// to copy over the tmate binary. Internal steps are not visible
+		// to the end user.
+		spec.Internal = append(spec.Internal, &engine.Step{
+			ID:         random(),
+			Pull:       engine.PullIfNotExists,
+			Image:      image.Expand(c.Tmate.Image),
+			Entrypoint: []string{"/bin/sh", "-c"},
```
Comment (bradrydzewski) [inline engine/compiler/compiler.go:449]: let's use the native `drone cp` command that is built-in to the runner.  The reason is the arm32 and arm64 images are scratch images and do not have the `cp` command available.

Example at: https://github.com/drone-runners/drone-runner-docker/blob/master/engine/compiler/compiler.go#L350:L351

File: engine/compiler/compiler.go [hunk 2]
```diff
@@ -427,6 +437,35 @@ func (c *Compiler) Compile(ctx context.Context, args runtime.CompilerArgs) runti
...
+		for _, step := range append(spec.Steps, spec.Internal...) {
+			step.Volumes = append(step.Volumes, &engine.VolumeMount{
+				Name: "_addons",
+				Path: "/usr/drone/bin",
+			})
+		}
+		spec.Volumes = append(spec.Volumes, &engine.Volume{
+			EmptyDir: &engine.VolumeEmptyDir{
+				ID:   random(),
+				Name: "_addons",
```
Comment (bradrydzewski) [inline engine/compiler/compiler.go:464]: can you add a linting rule that prevents a user from creating a volume named `_addons`?
https://github.com/drone-runners/drone-runner-docker/blob/master/engine/linter/linter.go#L160

**Comment by bradrydzewski** [comment]
this is great 🎉   I added a few comments to the code.  Also, one last piece is to add support for self-hosted tmate servers:
https://github.com/drone-runners/drone-runner-docker/commit/00b690ce8c9aa020b816bc47dae3be9fdef89294

File: engine/compiler/compiler.go (hunk 2 above)
Comment (shubham149) [reply engine/compiler/compiler.go:476]: Done

File: engine/compiler/compiler.go (hunk 1 above)
Comment (shubham149) [reply engine/compiler/compiler.go]: Done

**Comment by shubham149** [comment]
Addressed the comments

**Pull request merged**